src/core/pyproxy.c

```diff
@@ -209,18 +204,24 @@ pyproxy_getflags(PyObject* pyobj)
   // YIELD_FROM. GET_AWAITABLE uses _PyCoro_GetAwaitableIter defined in
   // genobject.c. This tests whether _PyCoro_GetAwaitableIter is likely to
   // succeed.
-  SET_FLAG_IF(IS_AWAITABLE, async_proto->am_await || gen_is_coroutine(pyobj));
+  //  || gen_is_coroutine(pyobj)
```

Comment:
What's this?

Reply:
Oh I left it here as a note to stick `gen_is_coroutine` outside of this function -- this function now only works on the type, but `gen_is_coroutine` needs to look at `generator->gen_code->co_flags` to see whether `CO_COROUTINE` is set, so it depends on the object itself not just the type of the object. I moved this down to then end of `pyproxy_getflags`.

Reply:
Thanks for the explanation. It would be good to add a comment about it.